--- buffer.h.orig	2026-08-17 12:12:31
+++ buffer.h	2026-08-17 12:14:06
@@ -27,9 +27,9 @@
 #ifndef TINYSTL_BUFFER_H
 #define TINYSTL_BUFFER_H
 
-#include <TINYSTL/allocator.h>
-#include <TINYSTL/new.h>
-#include <TINYSTL/traits.h>
+#include <tinystl/allocator.h>
+#include <tinystl/new.h>
+#include <tinystl/traits.h>
 
 namespace tinystl {
 
@@ -203,7 +203,7 @@
 	static inline T* buffer_insert_common(buffer<T, Alloc>* b, T* where, size_t count) {
 		const size_t offset = (size_t)(where - b->first);
 		const size_t newsize = (size_t)((b->last - b->first) + count);
-		if (b->first + newsize > b->capacity)
+		if (!b->first || b->first + newsize > b->capacity)
 			buffer_reserve(b, (newsize * 3) / 2);
 
 		where = b->first + offset;
