--- hash_base.h.orig	2026-08-16 23:29:08
+++ hash_base.h	2026-08-16 23:31:06
@@ -94,9 +94,9 @@
 	}
 
 	template<typename Key, typename Value>
-	static inline pair<typename remove_reference<Key>::type, typename remove_reference<Value>::type>
+	static inline pair<typename remove_const_reference<Key>::type, typename remove_const_reference<Value>::type>
 	make_pair(Key&& key, Value&& value) {
-		return pair<typename remove_reference<Key>::type, typename remove_reference<Value>::type>(
+		return pair<typename remove_const_reference<Key>::type, typename remove_const_reference<Value>::type>(
 				  static_cast<Key&&>(key)
 				, static_cast<Value&&>(value)
 			);
