--- string.h.orig	2026-08-17 12:12:31
+++ string.h	2026-08-17 12:14:06
@@ -27,9 +27,9 @@
 #ifndef TINYSTL_STRING_H
 #define TINYSTL_STRING_H
 
-#include <TINYSTL/allocator.h>
-#include <TINYSTL/stddef.h>
-#include <TINYSTL/hash.h>
+#include <tinystl/allocator.h>
+#include <tinystl/stddef.h>
+#include <tinystl/hash.h>
 
 namespace tinystl {
 
@@ -66,7 +66,7 @@
 		pointer m_capacity;
 
 		static const size_t c_nbuffer = 12;
-		char m_buffer[12];
+		char m_buffer[12]{0};
 	};
 
 	template<typename allocator>
@@ -157,7 +157,8 @@
 	template<typename allocator>
 	inline size_t basic_string<allocator>::size() const
 	{
-		return (size_t)(m_last - m_first);
+		ptrdiff_t size = m_last - m_first;
+		return (size_t)size;
 	}
 
 	template<typename allocator>
@@ -165,7 +166,7 @@
 		if (m_first + capacity + 1 <= m_capacity)
 			return;
 
-		const size_t size = (size_t)(m_last - m_first);
+		const ptrdiff_t size = m_last - m_first;
 
 		pointer newfirst = (pointer)allocator::static_allocate(capacity + 1);
 		for (pointer it = m_first, newit = newfirst, end = m_last; it != end; ++it, ++newit)
@@ -180,9 +181,9 @@
 
 	template<typename allocator>
 	inline void basic_string<allocator>::resize(size_t size) {
-		const size_t prevSize = m_last-m_first;
+		const ptrdiff_t prevSize = m_last-m_first;
 		reserve(size);
-		if (size > prevSize)
+		if (size > (size_t)prevSize)
 			for (pointer it = m_last, end = m_first + size + 1; it < end; ++it)
 				*it = 0;
 		else if (m_last != m_first)
@@ -198,7 +199,7 @@
 
 	template<typename allocator>
 	inline void basic_string<allocator>::append(const char* first, const char* last) {
-		const size_t newsize = (size_t)((m_last - m_first) + (last - first) + 1);
+		const ptrdiff_t newsize = ((m_last - m_first) + (last - first) + 1);
 		if (m_first + newsize > m_capacity)
 			reserve((newsize * 3) / 2);
 
@@ -217,12 +218,12 @@
 	inline void basic_string<allocator>::shrink_to_fit() {
 		if (m_first == m_buffer) {
 		} else if (m_last == m_first) {
-			const size_t capacity = (size_t)(m_capacity - m_first);
+			const ptrdiff_t capacity = m_capacity - m_first;
 			if (capacity)
 				allocator::static_deallocate(m_first, capacity+1);
 			m_capacity = m_first;
 		} else if (m_capacity != m_last) {
-			const size_t size = (size_t)(m_last - m_first);
+			const ptrdiff_t size = m_last - m_first;
 			char* newfirst = (pointer)allocator::static_allocate(size+1);
 			for (pointer in = m_first, out = newfirst; in != m_last + 1; ++in, ++out)
 				*out = *in;
@@ -236,35 +237,29 @@
 
 	template<typename allocator>
 	inline void basic_string<allocator>::swap(basic_string& other) {
-		const pointer tfirst = m_first, tlast = m_last, tcapacity = m_capacity;
-		m_first = other.m_first, m_last = other.m_last, m_capacity = other.m_capacity;
-		other.m_first = tfirst, other.m_last = tlast, other.m_capacity = tcapacity;
-
-		char tbuffer[c_nbuffer];
-
-		if (m_first == other.m_buffer)
-			for  (pointer it = other.m_buffer, end = m_last, out = tbuffer; it != end; ++it, ++out)
-				*out = *it;
-
-		if (other.m_first == m_buffer) {
-			other.m_last = other.m_last - other.m_first + other.m_buffer;
-			other.m_first = other.m_buffer;
-			other.m_capacity = other.m_buffer + c_nbuffer;
-
-			for (pointer it = other.m_first, end = other.m_last, in = m_buffer; it != end; ++it, ++in)
-				*it = *in;
-			*other.m_last = 0;
+		{
+			const pointer tfirst = m_first, tlast = m_last, tcapacity = m_capacity;
+			m_first = other.m_first, m_last = other.m_last, m_capacity = other.m_capacity;
+			other.m_first = tfirst, other.m_last = tlast, other.m_capacity = tcapacity;
 		}
 
+		for (size_t i = 0; i < c_nbuffer; ++i) {
+			const char temp = m_buffer[i];
+			m_buffer[i] = other.m_buffer[i];
+			other.m_buffer[i] = temp;
+		}
 		if (m_first == other.m_buffer) {
-			m_last = m_last - m_first + m_buffer;
+			ptrdiff_t len = m_last - m_first;
 			m_first = m_buffer;
+			m_last = m_buffer + len;
 			m_capacity = m_buffer + c_nbuffer;
-
-			for (pointer it = m_first, end = m_last, in = tbuffer; it != end; ++it, ++in)
-				*it = *in;
-			*m_last = 0;
 		}
+		if (other.m_first == m_buffer) {
+			ptrdiff_t len = other.m_last - other.m_first;
+			other.m_first = other.m_buffer;
+			other.m_last = other.m_buffer + len;
+			other.m_capacity = other.m_buffer + c_nbuffer;
+		}
 	}
 
 	template<typename allocatorl, typename allocatorr>
