--- unordered_set.h.orig	2026-08-16 23:29:08
+++ unordered_set.h	2026-08-16 23:31:37
@@ -78,7 +78,6 @@
 		: m_size(0)
 	{
 		buffer_init<pointer, Alloc>(&m_buckets);
-		buffer_resize<pointer, Alloc>(&m_buckets, 9, 0);
 	}
 
 	template<typename Key, typename Alloc>
@@ -89,10 +88,12 @@
 		buffer_init<pointer, Alloc>(&m_buckets);
 		buffer_resize<pointer, Alloc>(&m_buckets, nbuckets, 0);
 
-		for (pointer it = *other.m_buckets.first; it; it = it->next) {
-			unordered_hash_node<Key, void>* newnode = new(placeholder(), Alloc::static_allocate(sizeof(unordered_hash_node<Key, void>))) unordered_hash_node<Key, void>(*it);
-			newnode->next = newnode->prev = 0;
-			unordered_hash_node_insert(newnode, hash(it->first), m_buckets.first, nbuckets - 1);
+		if (other.m_buckets.first) {
+			for (pointer it = *other.m_buckets.first; it; it = it->next) {
+				unordered_hash_node<Key, void>* newnode = new(placeholder(), Alloc::static_allocate(sizeof(unordered_hash_node<Key, void>))) unordered_hash_node<Key, void>(*it);
+				newnode->next = newnode->prev = 0;
+				unordered_hash_node_insert(newnode, hash(it->first), m_buckets.first, nbuckets - 1);
+			}
 		}
 	}
 
@@ -125,9 +126,13 @@
 
 	template<typename Key, typename Alloc>
 	inline typename unordered_set<Key, Alloc>::iterator unordered_set<Key, Alloc>::begin() const {
-		iterator cit;
-		cit.node = *m_buckets.first;
-		return cit;
+		iterator it;
+		if (m_buckets.first) {
+			it.node = *m_buckets.first;
+		} else {
+			it.node = 0;
+		}
+		return it;
 	}
 
 	template<typename Key, typename Alloc>
@@ -149,13 +154,15 @@
 
 	template<typename Key, typename Alloc>
 	inline void unordered_set<Key, Alloc>::clear() {
-		pointer it = *m_buckets.first;
-		while (it) {
-			const pointer next = it->next;
-			it->~unordered_hash_node<Key, void>();
-			Alloc::static_deallocate(it, sizeof(unordered_hash_node<Key, void>));
+		if (m_buckets.first) {
+			pointer it = *m_buckets.first;
+			while (it) {
+				const pointer next = it->next;
+				it->~unordered_hash_node<Key, void>();
+				Alloc::static_deallocate(it, sizeof(unordered_hash_node<Key, void>));
 
-			it = next;
+				it = next;
+			}
 		}
 
 		m_buckets.last = m_buckets.first;
@@ -172,6 +179,7 @@
 
 	template<typename Key, typename Alloc>
 	inline void unordered_set<Key, Alloc>::rehash(size_t nbuckets) {
+		if (!m_buckets.first) return;
 		if (m_size + 1 > 4 * nbuckets) {
 			pointer root = *m_buckets.first;
 
@@ -201,6 +209,7 @@
 		unordered_hash_node<Key, void>* newnode = new(placeholder(), Alloc::static_allocate(sizeof(unordered_hash_node<Key, void>))) unordered_hash_node<Key, void>(key);
 		newnode->next = newnode->prev = 0;
 
+		if (!m_buckets.first) buffer_resize<pointer, Alloc>(&m_buckets, 9, 0);
 		const size_t nbuckets = (size_t)(m_buckets.last - m_buckets.first);
 		unordered_hash_node_insert(newnode, hash(key), m_buckets.first, nbuckets - 1);
 
@@ -225,6 +234,7 @@
 		unordered_hash_node<Key, void>* newnode = new(placeholder(), Alloc::static_allocate(sizeof(unordered_hash_node<Key, void>))) unordered_hash_node<Key, void>(static_cast<Key&&>(key));
 		newnode->next = newnode->prev = 0;
 
+		if (!m_buckets.first) buffer_resize<pointer, Alloc>(&m_buckets, 9, 0);
 		const size_t nbuckets = (size_t)(m_buckets.last - m_buckets.first);
 		unordered_hash_node_insert(newnode, keyhash, m_buckets.first, nbuckets - 1);
 
